analyses/hotspots-detection/utils/filterMaf.R

```diff
@@ -0,0 +1,141 @@
+                      impact_values,
+                      hotspot_database_2017_snv_df,
+                      hotspot_database_2017_indel_df,
+                      hotspot_genomic_site_df) {
+  if (!missing(impact_values)) {
+    # IMPACT filtering values
+    table <- table %>%
+      dplyr::filter(IMPACT %in% impact_values)
+  }
+  
+  # filtered calls_base
+  calls_base <- table %>%
+    # only keep annotation for canonical protein_coding transcripts 
+    filter( CANONICAL=='YES' & 
+              BIOTYPE=='protein_coding') %>%
+    as.data.frame() %>%
+    # create an Amino_Acid_Position column
+    dplyr::mutate(
+      Amino_Acid_Position =
+        case_when(
+          # If not splice site we want to capture Amino_Acid_Position to match
+          !grepl("Splice_Site",Variant_Classification) ~
+            # format of Protein_position is 594/766
+            # [Amino_Acid_Position/ Protein_length]
+            # stripping protein length here to get Amino_Acid_Position
+            gsub("/.*", "", Protein_position),
+          # If splice site we want to capture HGVSp_Short to match 
+          grepl("Splice_Site",Variant_Classification) ~
+            # since these sites are intronic sites adding
+            # HGSVp_Short instead of Protein_position
+            # in the format of `X000_splice`
+            gsub("p[.]","",HGVSp_Short))
+    ) 
+
+  # stop if both SNV+splice and INDEL hotspot is not provided as MSKCC hotspot database
+  if (!missing(hotspot_database_2017_snv_df) & missing(hotspot_database_2017_indel_df)){
+    stop("Provide complete mskcc hotspot; Indel hotspot is missing ")
+  } else if (missing(hotspot_database_2017_snv_df) & !missing(hotspot_database_2017_indel_df)){
+    stop("Provide complete mskcc hotspot; SNV+ splice hotspot is missing ")
+  }
+  
+  
+  if (!missing(hotspot_database_2017_snv_df) & !missing(hotspot_database_2017_indel_df)) {
+    # filter by amino acid or HGSVp_Short values
+    # for splice sites by matching Amino_Acid_Position in
+    # hotspot_database_2017_snv_df
+    calls_base_aa_filt_match <- calls_base %>%
+      dplyr::inner_join(hotspot_database_2017_snv_df,
+        by = c("Amino_Acid_Position", "Hugo_Symbol")
+      ) %>%
+      select(-Amino_Acid_Position)
+    
+    # if indel then check for overlap 
+    # within region of Amino_Acid_Position
+    calls_base_aa_filt_indel <- calls_base %>%
+      # if indel in Variant_Classification
+      filter(grepl("Ins|Del",Variant_Classification )) %>%
+      mutate( Amino_Acid_Start = gsub("-.*" , "", Amino_Acid_Position),
+              Amino_Acid_End = gsub ("*.-" , "" , Amino_Acid_Position)
+              ) %>%
+      dplyr:: left_join(hotspot_database_2017_indel_df,
+                        by = c( "Hugo_Symbol"),
+                        suffix = c(".calls_base",".hotspot_indel_df")  
+      ) %>% 
+      # keep only sites that where 
+      # calls$start >= hotspot_indel$start and
+      # calls$end <= hotspot_indel$stop
```

Comment:
Updating this comment as it didn't match the code after discussion & updates.
```suggestion
      # keep only sites that overlap hotspot indels with  
      # calls$start <= hotspot_indel$stop and
      # calls$end >= hotspot_indel$start
```